src/MissionManager/PlanMasterController.cc

```diff
@@ -212,6 +215,17 @@ void PlanMasterController::_activeVehicleChanged(Vehicle* activeVehicle)
     _updatePlanCreatorsList();
 }
 
+void PlanMasterController::_managerError(int errorCode, const QString& errorMsg)
+{
+    _missionController.abortSync();
+    _missionController.setSkipPlanView(true);
+    _geoFenceController.abortSync();
+    _geoFenceController.setSkipPlanView(true);
+    _rallyPointController.abortSync();
+    _rallyPointController.setSkipPlanView(true);
+    removeAllFromVehicle();    
```

Comment:
This would also remove the plan if a download fails. This is probably not desired.

Reply:
Hm that is true, and is probably not desired. The logic for upload and download is very intertwined so I will try to figure out how to only remove it for upload

Reply:
Do you think the feature of removing the mission/rallypoint/geofence from the drone when upload fails is important? Implementing it into the current code is quite hard

Reply:
I do not think so, it is much more important to block start of a failed upload, and visualise what is wrong.

Reply:
I agree. The only dangerous aspect of not removing the mission that is partially uploaded due to an error, is that if the user restarts QGC, QGC will download the mission from the drone, not knowing it was partially uploaded. This is a case for leaving the mission initially downloaded on QGC as dirty, as discussed in Slack

Reply:
This is fixed by not downloading mission from drone on QGC startup